seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json

```diff
@@ -1180,10 +1180,12 @@
         "var myStr = \"FirstLine\\n\\t\\\\SecondLine\\nThirdLine\";"
       ],
       "tests": [
-        "assert(myStr === \"FirstLine\\n\\t\\\\SecondLine\\nThirdLine\", 'message: <code>myStr</code> should have encoded text with the proper escape sequences and no spacing.');",
-        "assert(myStr.match(/\\n/g).length == 2, 'message: <code>myStr</code> should have two newline characters <code>\\n</code>');",
-        "assert(myStr.match(/\\t/g).length == 1, 'message: <code>myStr</code> should have one tab character <code>\\t</code>');",
-        "assert(myStr.match(/\\\\/g).length == 1, 'message: <code>myStr</code> should have one correctly escaped backslash character <code>\\\\</code>');"
+        "assert(myStr.match(/ /g) === null, 'message: <code>myStr</code> should not contain any spaces');",
+        "assert((myStr.match(/FirstLine/g).length == 1) && (myStr.match(/SecondLine/g).length == 1) && (myStr.match(/ThirdLine/g).length == 1), 'message: <code>myStr</code> should contain the strings <code>FirstLine</code>, <code>SecondLine</code> and <code>ThirdLine</code> (remember case sensitivity!)');",
```

Comment:
The assertion part of this line could be slightly shorter if we used the `RegEx.test()` method. It returns a boolean telling whether or not it found a match.

Would you mind changing this
`(myStr.match(/FirstLine/g).length == 1) && (myStr.match(/SecondLine/g).length == 1) && (myStr.match(/ThirdLine/g).length == 1)`

...to the following? :blush:
`/FirstLine/.test(myStr) && /SecondLine/.test(myStr) && /ThirdLine/.test(myStr)`

After testing this for all test cases, it seems like `g` flag isn't necessary to make the tests work - probably since we're only looking for a single occurrence :open_mouth: